src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -98,10 +102,19 @@ private static (List<Error>, TableOfContentsModel) LoadTocModel(JToken tocToken)
 
             var (errors, model) = LoadTocModel(context, file, content);
 
+            if (file == rootPath)
```

Comment:
Can this section be lifted outside this recursive function? It makes it easier to see what are applied to every node vs what else we do

Reply:
filemoniker can only be got after the metadata has been extract

---
In reply to: [292299489](https://github.com/dotnet/docfx/pull/4702#discussion_r292299489) [](ancestors = 292299489)

Reply:
ok, then can this be done after #4711 ?